sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosVectorDistanceFunction.java

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+
+import java.util.Arrays;
+
+/**
+ * Distance Function for the embeddings in the Cosmos DB database service.
+ */
+public enum CosmosVectorDistanceFunction {
```

Comment:
Will the `enum`s added in this PR not grow over time? - if the customer uses ordinals to refer to an `enum` type then that could break the customer's code if new `enum` types are added. We went with creating singleton instances like in `CosmosRegionSwitchHint`.